src/main/src/main/java/org/geoserver/config/GeoServerLoader.java

```diff
@@ -206,6 +219,8 @@ public void reload() throws Exception {
         
         loadCatalog( catalog, xp );
         loadGeoServer( geoserver, xp);
+        
+        reloadInitializers(geoserver);
```

Comment:
That makes me a little uncomfortable. Is it worth clearing the ResourcePool after the initializers run? Or making the ResourcePoolInitalizer clear the ResourcePool datastore cache when it is done.